src/plugins/discover/public/plugin.ts

```diff
@@ -62,6 +64,13 @@ export interface DiscoverStart {
      */
     DocViewer: React.ComponentType<DocViewRenderProps>;
   };
+  savedSearches: {
+    /**
+     * Create a {@link SavedObjectLoader | loader} to handle the saved searches type.
+     * @param services
+     */
+    createLoader(services: SavedObjectKibanaServices): SavedObjectLoader;
+  };
```

Comment:
While trying to reduce the initial bundle size, I saw that the async `import` of the management section mount I added was only reducing the SOM bundle size from 4mb to 3.7mb. After a few check, the cause was the `createSavedSearchesLoader` static import I'm using in the SOM plugin, that caused the full `discover` plugin + deps to be bundled into the SOM plugin.

After exposing this method from the contract instead, the SOM plugin is no longer using non-type imports from any plugin, and the initial bundle size went from 3.8mb to 51kb.

cc @elastic/kibana-app (btw it seemsthe `discover` NP plugin is not present in `CODEOWNERS`)

Reply:
Thanks for pinging about the codeowners, I will set up a PR for this.